no-downtime-upgrade/src/main/java/org/keycloak/playground/nodowntimeupgrade/naive_jackson/ObjectEntity_V3.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.playground.nodowntimeupgrade.naive_jackson;
+
+import org.keycloak.playground.nodowntimeupgrade.base.model.ObjectAdapter_V3;
+import org.keycloak.playground.nodowntimeupgrade.base.model.ModelVersion;
+import org.keycloak.playground.nodowntimeupgrade.base.model.ObjectModel_V3;
+import static org.keycloak.playground.nodowntimeupgrade.base.model.Constants.DEFAULT_V3_TIMEOUT;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public class ObjectEntity_V3 {
+
+    public int entityVersion = ModelVersion.VERSION_3.getVersion();
```

Comment:
It seems the "entityVersion" is currently required field on the entities based on the AbstractNaiveVersionedJacksonStorage implementation. Perhaps some better "type safety" can be used to avoid some errors in case that implementor forgot to implement "entityVersion" field on the entity objects? Like the abstract subclass (or interface), which the entity objects will inherit from and which will ensure that "entityVersion" field is present.

Reply:
True. Purpose of the `naive_jackson` is to provide naive working implementation, not the final solution - that will be left for proponents